src/main/resources/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote/script.js

```diff
@@ -14,9 +14,17 @@ Behaviour.specify("span.pipeline-new-node", 'NewNodeConsoleNote', 0, function(e)
     var nodeId = e.getAttribute('nodeId')
     var startId = e.getAttribute('startId')
     if (startId == null || startId == nodeId) {
-        e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" onclick="showHidePipelineSection(this); return false">hide</a>)</span>')
+        e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" class="pipeline-toggle">hide</a>)</span>')
         // TODO automatically hide second and subsequent branches: namely, in case a node has the same parent as an earlier one
+
+        const toggle = e.querySelector('.pipeline-show-hide:last-child .pipeline-toggle');
```

Comment:
Prefer  `querySelectorAll` to reliably select the last `.pipeline-toggle` element, as it works regardless of the DOM structure:

```suggestion
        const toggles = e.querySelectorAll('.pipeline-show-hide .pipeline-toggle');
        const toggle = toggles[toggles.length - 1];
```

Picking the last item in the list ensures the correct element is always selected, even if `e` contains multiple `.pipeline-show-hide elements`. In contrast, `querySelector('.pipeline-show-hide:last-child .pipeline-toggle')` only works if `.pipeline-show-hide` is the last child of its parent, making it fragile if the DOM changes.

Reply:
Understood, sorry for the back and forth with this change. 

I appreciate your explanation and will note it moving forward.